tests/31sync/09archived.pl

```diff
@@ -121,6 +121,69 @@
    };
 
 
+test "Oldly left rooms don't appear in the leave section of sync",
+   requires => [ local_user_fixture( with_events => 0 ), local_user_fixture( with_events => 0 ),
+                 qw( can_sync ) ],
+
+   bug => 'SYN-589',
+
+   check => sub {
+      my ( $user, $user2 ) = @_;
+
+      my ( $filter_id, $room_id_1, $room_id_2, $next );
+
+      my $filter = {
+         room => { timeline => { limit => 1 }, include_leave => JSON::true }
+      };
+
+      matrix_create_filter( $user, $filter )->then( sub {
+         ( $filter_id ) = @_;
+
+         Future->needs_all(
+            matrix_create_room( $user )->on_done( sub { ( $room_id_1 ) = @_; } ),
+            matrix_create_room( $user )->on_done( sub { ( $room_id_2 ) = @_; } ),
+         );
+      })->then( sub {
+         matrix_join_room( $user2, $room_id_1 );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         $next = $body->{next_batch};
+
+          Future->done(1);
+      })->then( sub {
+         matrix_leave_room( $user, $room_id_1 );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id );
```

Comment:
Did you mean to provide a `since` argument here? Otherwise, you're just about to overwrite the `$next` variable, having not actually used it since it was written first on line 153.